src/typing_extensions.py

```diff
@@ -813,6 +813,15 @@ def _typeddict_new(*args, total=True, **kwargs):
             raise TypeError("TypedDict takes either a dict or keyword arguments,"
                             " but not both")
 
+        if kwargs:
+            warnings.warn(
```

Comment:
I think it's important that we copy CPython's behaviour here, and Cpython emits a deprecation warning here on 3.11+. @JelleZijlstra do you think we should only emit the deprecation warning if the user is running typing_extensions on 3.11+?

Reply:
That's a hard question and gets into what we would do if we ever want to break backwards compatibility for typing-extensions. My first instinct is to say we should always warn in typing-extensions, regardless of the version, but then what would we do when CPython removes support for kwargs-based TypedDicts? I'd be hesitant to remove the runtime behavior in typing-extensions and break backwards compatibility.

Reply:
I agree it's a hard question.

Elsewhere in the `typing_extensions` implementation of `TypedDict`, we actually already have two places where CPython  long ago removed support for a certain feature, but `typing_extensions` has in effect had "eternal deprecation warnings":

https://github.com/python/typing_extensions/blob/fb37b2ee0ab56baf2dfd5df27aec0430435ba17f/src/typing_extensions.py#L789-L791

https://github.com/python/typing_extensions/blob/fb37b2ee0ab56baf2dfd5df27aec0430435ba17f/src/typing_extensions.py#L804-L806

Reply:
I think eternal deprecation warnings might be the right answer, actually. (Or eternal until we ever make typing-extensions 5 in the distant uncertain future.)

Reply:
So maybe we do want the change in https://github.com/python/typing_extensions/pull/150#discussion_r1174612309? ;)

Reply:
Yes, let's do that. I commented there because the change looked unrelated to this PR.